### DISCUSSION TITLE
cubic crystal symmetry

### DISCUSSION THREAD

**Discussion by anujyadav0693** [neperfepx/neper#593] [Q&A] [answered]
sir,
i want to made a micro structure with random orientation in cubic symmetry. this code is write or wrong please tell
neper -T -n 100 -dim 2 -domain "square(0.01,0.04)" -oridescriptor quaternion -ori random  -oricrysym cubic -statcell diameq -o 2d100grains  -format geo,inp,tess,ori 
we can use -ori random and -oricrysysm together.
thank you.

--

sir one more question
after giving -oricrysysm cubic both -ori random/uniform gives approximately same ori file why?
what is difference between random and uniform orientation in cubic symmetry?
thank you

**Comment by rquey** [comment, marked as answer]
Your code is right.

The orientations should not be the same when you use `-ori random` or `-ori uniform`. They may be relatively close (or not totally disconnected) because a uniform orientation distribution is generated using the random orientation distribution as initial state.

See [this paper](https://neper.info/papers.html#id3) for what `uniform` does.
